src/main/java/com/fasterxml/jackson/core/base/ParserBase.java

```diff
@@ -1553,12 +1554,16 @@ protected ContentReference _contentReferenceRedacted() {
         return ContentReference.redacted();
     }
 
-    protected static int[] growArrayBy(int[] arr, int more)
+    protected static int[] growArrayBy(int[] arr, int more) throws StreamConstraintsException
```

Comment:
I don't like use of API level `StreamConstraintsException` at low-level helper methods; just make it `IllegalArgumentException`

Reply:
As per my comment on issue, we can change it back to `StreamConstraintsException` if we absolutely must (run out of time for other checks). Just not yet.